tap_github/client.py

```diff
@@ -243,6 +243,13 @@ def post_process(self, row: dict, context: Optional[Dict[str, str]] = None) -> d
             row["repo_id"] = context["repo_id"]
         return row
 
+    def backoff_handler(self, details: dict) -> None:
+        """Handle retriable error by swapping auth token."""
+        self.logger.info("Retrying request with different token")
```

Comment:
This is only for 403, no? Other retriable error do not need to this.

Reply:
Yes, we only need this for rate limit errors, but as I mentioned in the PR, I didn't find a way to easily filter errors. Ideas welcome :)

Reply:
Sorry I looked to quickly. And we don't have access to the status code on the backoff handler? I feel like this would be a good addition for this function in the SDK. Maybe we can merge like this and open an issue on the SDK and our tap?

Reply:
Yes, I think it would make sense to have some way of knowing why the backoff was needed. The reason would be determined in `validate_response`, and an easy fix could be to set a value on the stream object, which this handler could read and reset, but it feels a little hacky. Or is it possible to attach some value to the `RetriableAPIError` object? but then I'm not even sure the handler actually gets the exception object anyway.